src/Database/Persist/URL.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Database.Persist.URL
+    ( fromDatabaseUrl
+    ) where
+
+import Control.Monad (MonadPlus, mzero)
+import Data.ByteString (ByteString, uncons)
+import Data.Monoid ((<>))
+import Database.Persist.Postgresql (PostgresConf(..))
+import URI.ByteString
+    ( Authority(..)
+    , Host(..)
+    , Port(..)
+    , URI(..)
+    , UserInfo(..)
+    , parseURI
+    , strictURIParserOptions
+    )
+
+import qualified Data.ByteString.Char8 as Char8
+
+-- | Build a @'PostgresConf'@ by parsing a database URL String
+fromDatabaseUrl :: MonadPlus m => Int -> ByteString -> m PostgresConf
```

Comment:
Gotcha, that seems reasonable. When I was doing this I instantiated to `Either` then `Parser` in the `parseJSON` definitions, which isn't super great I guess.